src/modules/hip/kernel/tensor_min.hpp

```diff
@@ -0,0 +1,466 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 - Reduction Stage 2 --------------------
+
+template <typename T>
+__global__ void tensor_min_grid_3channel_result_hip(float *srcPtr,
+                                                uint xBufferLength,
+                                                T *dstPtr)
+{
+    int id_x = hipThreadIdx_x * 8;
+    int id_z = hipBlockIdx_z;
+
+    __shared__ float partialRMin_smem[256];                           // 1024 floats of src reduced to 256 in a 256 x 1 thread block
+    __shared__ float partialGMin_smem[256];                           // 1024 floats of src reduced to 256 in a 256 x 1 thread block
+    __shared__ float partialBMin_smem[256];                           // 1024 floats of src reduced to 256 in a 256 x 1 thread block
+
+    uint srcIdx = (id_z * xBufferLength) * 3;
+    float srcRefR = srcPtr[srcIdx];                                 // get starting value of R channel
+    float srcRefG = srcPtr[srcIdx + 1];                             // get starting value of G channel
+    float srcRefB = srcPtr[srcIdx + 2];                             // get starting value of B channel
+
+    partialRMin_smem[hipThreadIdx_x] = srcRefR;                       // initialization of LDS for R channel to srcRefR using all 256 x 1 threads
+    partialGMin_smem[hipThreadIdx_x] = srcRefG;                       // initialization of LDS for G channel to srcRefG using all 256 x 1 threads
+    partialBMin_smem[hipThreadIdx_x] = srcRefB;                       // initialization of LDS for B channel to srcRefB using all 256 x 1 threads
+
+    if (id_x >= xBufferLength)
+        return;
+
+    int xAlignedLength = xBufferLength & ~7;                        // alignedLength for vectorized global loads
+    int xDiff = xBufferLength - xAlignedLength;                     // difference between bufferLength and alignedLength
+    srcIdx += id_x * 3;
+
+    d_float24 src_f24;
+    rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(srcPtr + srcIdx, &src_f24);           // load 24 pixels to local mmemory
+    if (id_x + 8 > xBufferLength)
```

Comment:
not sure what is this loop for

Reply:
If adding 8 to the 8 element length vectorized id_x, goes > bufferLength, then we can't accept for the remaining values in the row while calculating min. So gpu local memory vector is reset to the reference min.

Reply:
I will adjust srcIdx as discussed.

Reply:
clarified